### PR TITLE
merge up v3.8.0

### LINKED ISSUE #26604: [Doc]: Inappropriate example in gallery
### Documentation Link

https://matplotlib.org/stable/gallery/statistics/histogram_features.html

### Problem

* Given the sordid historical misuse of IQ, this gallery example unnecessarily reinforces a crude association between IQ and "smarts" that distracts from the main point, which is to show how to make a histogram.
* Any attempt to contextualize the meaning of IQ would further distract from this point, so it would be better to just change the axis labels to avoid the negative association.

### Suggested improvement

* Remove the axis labels altogether, as was done [here](https://matplotlib.org/stable/gallery/statistics/hist.html#histograms), or change the simulation example to something more neutral, such as the masses of apples from a hypothetical farm.

### LINKED ISSUE #26736: [ENH]: matplotlib 3.12 prelease wheels
### Problem

Now that #26647 is merged, would it be possible to have a matplotlib 3.8.0rc2 release with 3.12 wheels soon. It would be helpful for folks (like me) trying to test downstream (from matplotlib) packages against 3.12.

### Proposed solution

_No response_

### LINKED ISSUE #26738: Improve LineCollection docstring further
(M, 2)

I would perhaps completely drop the "list of points" and just write
```
A sequence ``[line0, line1, ...]`` where each line is a (N, 2)-shape
array-like of points::

    line0 = [(x0, y0), (x1, y1), ...]

Each line can...
```

_Originally posted by @anntzer in https://github.com/matplotlib/matplotlib/pull/26676#discussion_r1313026557_

### PR DESCRIPTION
**Pull request by ksunden**
Merge up post release 3.8.0

- Update github stats for v 3.8.0rc1
- REL: v3.8.0rc1
- Bump off of release tag for v3.8.0rc1
- Backport PR #26487: DOC: Remove unused image rotator
- Backport PR #26498: Add plausible analytics to the documentation pages
- Backport PR #26502: TST: Increase some tolerances for non-x86 arches
- Backport PR #26490: Import PIL.Image explicitly over PIL
- Backport PR #26173: Synchronize mathtext docs and handling
- Backport PR #26504: TYP: Add overload to specify output of Colormap.__call__ when possible
- Backport PR #26509: Update/tweak SpanSelector docs.
- Backport PR #26526: Bump pypa/cibuildwheel from 2.14.1 to 2.15.0
- Backport PR #26513: Tweak shape repr in _api.check_shape error message.
- Backport PR #26532: Fix input check in Poly3DCollection.__init__
- Backport PR #26529: Fix MathText antialiasing
- Backport PR #26201: DOC: Add documentation on codespaces usage
- Backport PR #26493: Disable ``add_html_cache_busting`` on Sphinx 7.1+
- Backport PR #26514: Clarify interaction between params of get_path_collection_extents.
- Backport PR #26545: Fix size inferral when using cairocffi
- Backport PR #26554: Remove NumPy abs overrides from pylab
- Backport PR #26519: Fix mathtext mismatched braces
- Backport PR #26543: Add ninja to Cygwin builder
- Backport PR #26462: Boxplot fix median line extending past box boundaries #19409
- Backport PR #26573: [DOC]: codespace link in contribute index
- Backport PR #26572: [DOC]: clarify pre-commits and editing workflow
- Backport PR #26578: MAINT: add __pycache__/ to .gitignore
- Backport PR #26581: Deduplicate test for toolbar button icon LA mode.
- Backport PR #26576: Use sys.platform over os.name
- Backport PR #26591: Fix ToolBase.figure property setter.
- Backport PR #26565: [doc]: added section Verify installation
- Backport PR #26606: [Doc] Revise histogram features example (Closes #26604)
- Backport PR #26614: Properly disconnect machinery when removing child axes.
- Backport PR #26598: FIX: array labelcolor for Tick
- Backport PR #24711: Test with Python 3.12
- Backport PR #26622: [Doc] Improve DSP-related examples
- Backport PR #26619: [DOC] Clarify some tick-related docstrings
- Backport PR #26540: TYP: Add overloads for FT2Font.get_sfnt_table
- Backport PR #26633: [Doc] Shorten documentation links in widgets
- Backport PR #24209: List the webagg_core module in the sphinx docs.
- Backport PR #26636: [Doc] Improve set_layout_engine docs
- Backport PR #26641: [Doc] Add ACCEPTS for some Axes set methods
- Backport PR #26582: MNT: Enable wheels for Python 3.12
- Backport PR #26521: Replaced list with tuple in pyplot for axes
- Backport PR #26635: [MNT] Do not configure axes properties via subplots(..., subplot_kw={...})
- Backport PR #26348: Test some untested Locator code
- Backport PR #26646: Use standard method for closing QApp when last window is closed.
- Backport PR #26597: Squeeze post-converted values when validating limits
- Backport PR #26649: [DOC] Remove "Discouraged" notices that have been superseded by deprecation
- Backport PR #26608: Removed unnecessary origin keywords
- Backport PR #26470: [DOC]: mathtext tutorial-consolidate explain and notes
- Backport PR #26656: TYP: Fix some small bugs
- Backport PR #26566: MAINT: Numpy 2.0 deprecations for row_stack and in1d
- Backport PR #26542: TST: Ensure test_webagg subprocess is terminated
- Backport PR #26245: [pre-commit.ci] pre-commit autoupdate
- Backport PR #26657: DOC: Fix some small issues
- Backport PR #26541: TYP: Add typing on mathtext internals
- Backport PR #26193: Sort tex2uni data in mathtext
- Backport PR #26665: Clarify loading of backend FigureCanvas and show().
- Backport PR #26680: Fix flaky CI tests
- Backport PR #26689: Fix error generation for missing pgf.texsystem.
- Backport PR #26695: Bump actions/checkout from 3 to 4
- Backport PR #26601: Avoid checking limits when updating both min and max for contours
- Backport PR #26709: DOC: consistency in docstrings of formatting of array-like
- Backport PR #26491: TYP: Add common-type overloads of subplot_mosaic
- Backport PR #26721: Add a Python 3.12 classifier
- Backport PR #26719: Fix issue with missing attribute in Path3DCollection
- Backport PR #26676: [DOC] Slightly improve the LineCollection docstring
- Backport PR #26671: [DOC] Enhance API reference index
- Backport PR #26705: [Doc] Small fixes found by velin
- Backport PR #26628: DOC: move install related FAQ to install docs
- Merge branch v3.7.x into v3.8.x
- Backport PR #26763: DOC: Add redirects for old gitwash files
- Backport PR #26700: Check type for set_clip_box
- DOC: Pin mpl-sphinx-theme to 3.8.x
- Backport PR #26762: MNT: Numpy 2.0 removals from ndarray class
- Backport PR #26767: Trim Gouraud triangles that contain NaN
- Backport PR #26687: Remove usage of recarray
- Backport PR #26702: converted coc to rst and put links in code_of_conduct.md
- Prepare for release v3.8.0
- REL: v3.8.0
- Bump post release v3.8.0
- Add zenodo svg for 3.8.0
- fix references which had ~ replaced by -
- Backport PR #26780: fix Axes.errorbar docstring

Closes #26736, though there could be discussions about in the future doing the merge up even for RCs so that the version number is bumped for nightlies

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by QuLogic** [comment]
> Closes #26738, though there could be discussions about in the future doing the merge up even for RCs so that the version number is bumped for nightlies

Is that the right issue? It doesn't appear to have anything to do with RCs/nightlies.

File: .circleci/config.yml [hunk 1]
```diff
@@ -100,8 +100,6 @@ commands:
             python -m pip install --user \
                 numpy<< parameters.numpy_version >> \
                 -r requirements/doc/doc-requirements.txt
-            python -m pip install --no-deps --user \
-                git+https://github.com/matplotlib/mpl-sphinx-theme.git
```
Comment (QuLogic) [inline .circleci/config.yml:104]: This should be kept.

File: lib/matplotlib/figure.pyi [hunk 2]
```diff
@@ -16,7 +16,7 @@ from matplotlib.backend_bases import (
 from matplotlib.colors import Colormap, Normalize
 from matplotlib.colorbar import Colorbar
 from matplotlib.cm import ScalarMappable
-from matplotlib.gridspec import GridSpec, SubplotSpec, SubplotParams as SubplotParams
```
Comment (QuLogic) [inline lib/matplotlib/figure.pyi:19]: This is accidentally reverting the move of `SubplotParams`, isn't it?

File: lib/matplotlib/figure.pyi [hunk 3]
```diff
@@ -28,6 +28,32 @@ from .typing import ColorType, HashableList
 
 _T = TypeVar("_T")
 
+class SubplotParams:
```
Comment (QuLogic) [inline lib/matplotlib/figure.pyi:31]: Ditto.

**Comment by ksunden** [comment]
> > Closes #26738, though there could be discussions about in the future doing the merge up even for RCs so that the version number is bumped for nightlies
> 
> Is that the right issue? It doesn't appear to have anything to do with RCs/nightlies.

Yeah, meant 26736, and apparently typed the wrong number, though that has actually been closed by the reporter already

File: .circleci/config.yml (hunk 1 above)
Comment (ksunden) [reply .circleci/config.yml]: fixed, this one didn't cause a merge conflict

File: lib/matplotlib/figure.pyi (hunk 2 above)
Comment (ksunden) [reply lib/matplotlib/figure.pyi]: Fixed, yeah, merge conflict and I apparently resolved in the wrong direction.

File: .circleci/config.yml [hunk 4]
```diff
@@ -102,7 +102,6 @@ commands:
                 -r requirements/doc/doc-requirements.txt
             python -m pip install --no-deps --user \
                 git+https://github.com/matplotlib/mpl-sphinx-theme.git
-
```
Comment (QuLogic) [inline .circleci/config.yml:105]: Extra change.

**Review by QuLogic — APPROVED** [review]

**Review by oscargus — APPROVED** [review]

**Pull request merged**